Source/Core/OrientedBoundingBox.js

```diff
@@ -0,0 +1,330 @@
+            rotMat = Matrix3.fromRotationZ(rotation, scratchRotation);
+        } else {
+            rotMat = Matrix3.clone(Matrix3.IDENTITY, scratchRotation);
+        }
+
+        var scale = scratchCartesian1;
+        scale.x = boundingRectangle.width * 0.5;
+        scale.y = boundingRectangle.height * 0.5;
+        scale.z = 0.0;
+        var scaleMat = Matrix3.fromScale(scale, scratchMatrix1);
+        Matrix3.multiply(rotMat, scaleMat, result.halfAxes);
+
+        var translation = Matrix3.multiplyByVector(rotMat, scale, result.center);
+        translation.x += boundingRectangle.x;
+        translation.y += boundingRectangle.y;
+
+        return result;
+    };
+
+    /**
+     * Duplicates a OrientedBoundingBox instance.
+     *
+     * @param {OrientedBoundingBox} box The bounding box to duplicate.
+     * @param {OrientedBoundingBox} [result] The object onto which to store the result.
+     * @returns {OrientedBoundingBox} The modified result parameter or a new OrientedBoundingBox instance if none was provided. (Returns undefined if box is undefined)
+     */
+    OrientedBoundingBox.clone = function(box, result) {
+        if (!defined(box)) {
+            return undefined;
+        }
+
+        if (!defined(result)) {
+            return new OrientedBoundingBox(box.center, box.halfAxes);
+        }
+
+        Cartesian3.clone(box.center, result.center);
+        Matrix3.clone(box.halfAxes, result.halfAxes);
+
+        return result;
+    };
+
+    /**
+     * Determines which side of a plane the oriented bounding box is located.
+     *
+     * @param {OrientedBoundingBox} box The oriented bounding box to test.
+     * @param {Cartesian4} plane The coefficients of the plane in Hessian Normal Form, as `ax + by + cz + d = 0`,
+     *                           where (a, b, c) must be a unit normal vector.
+     *                           The coefficients a, b, c, and d are the components x, y, z,
+     *                           and w of the {@link Cartesian4}, respectively.
+     * @returns {Intersect} {@link Intersect.INSIDE} if the entire box is on the side of the plane
+     *                      the normal is pointing, {@link Intersect.OUTSIDE} if the entire box is
+     *                      on the opposite side, and {@link Intersect.INTERSECTING} if the box
+     *                      intersects the plane.
+     */
+    OrientedBoundingBox.planeIntersect = function(box, plane) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(box)) {
+            throw new DeveloperError('box is required.');
+        }
+
+        if (!defined(plane)) {
+            throw new DeveloperError('plane is required.');
+        }
+        //>>includeEnd('debug');
+
+        var center = box.center;
+        // plane is used as if it is its normal; the first three components are assumed to be normalized
+        var radEffective = Math.abs(Cartesian3.dot(plane, Matrix3.getColumn(box.halfAxes, 0, scratchCartesian1)))
```

Comment:
Let's wait until we profile, but we could micro-optimize this a lot.

Also, make sure to profile for fps in a Cesium build with `DeveloperError` removed (`combineRelease` should do it I think), see https://github.com/AnalyticalGraphicsInc/cesium/wiki/Contributor%27s-Guide